crates/astria-core/src/primitive/v1/asset/denom.rs

```diff
@@ -207,90 +207,50 @@ impl TracePrefixed {
         self.trace.is_empty()
     }
 
-    /// Checks if the trace prefixed denom starts with `s`.
+    /// Checks if the trace prefixed denom has `port` in left-most position.
     ///
     /// # Examples
     ///
     /// ```
     /// use astria_core::primitive::v1::asset::denom::TracePrefixed;
     /// let denom = "four/segments/of/a/denom".parse::<TracePrefixed>().unwrap();
+    /// assert!(denom.has_leading_port("four"));
+    /// assert!(!denom.has_leading_port("segments"));
+    /// assert!(!denom.has_leading_port("of"));
+    /// assert!(!denom.has_leading_port("a"));
+    /// assert!(!denom.has_leading_port("denom"));
+    /// assert!(!denom.has_leading_port(""));
+    /// ```
+    #[must_use]
+    pub fn has_leading_port<T: AsRef<str>>(&self, port: T) -> bool {
```

Comment:
Added `has_leading_port` and `has_leading_channel` to have a stricter prefix check instead of `starts_with_str`.